src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSEKSResourceDetector.cs

```diff
@@ -0,0 +1,148 @@
+    /// </summary>
+    public class AWSEKSResourceDetector : IResourceDetector
+    {
+        private const string AWSEKSCertificatePath = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt";
+        private const string AWSEKSCredentialPath = "/var/run/secrets/kubernetes.io/serviceaccount/token";
+        private const string AWSEKSMetadataFilePath = "/proc/self/cgroup";
+        private const string AWSClusterInfoUrl = "https://kubernetes.default.svc/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info";
+        private const string AWSAuthUrl = "https://kubernetes.default.svc/api/v1/namespaces/kube-system/configmaps/aws-auth";
+
+        /// <summary>
+        /// Detector the required and optional resource attributes from AWS EKS.
+        /// </summary>
+        /// <returns>List of key-value pairs of resource attributes.</returns>
+        public IEnumerable<KeyValuePair<string, object>> Detect()
+        {
+            List<KeyValuePair<string, object>> resourceAttributes = null;
+
+            try
+            {
+                var clusterName = this.GetEKSClusterName();
+                var containerId = this.GetEKSContainerId();
+
+                if (clusterName == null && containerId == null)
+                {
+                    return resourceAttributes;
+                }
+
+                resourceAttributes = new List<KeyValuePair<string, object>>()
+                {
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudProvider, "aws"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudPlatform, "aws_eks"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeK8SClusterName, clusterName),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeContainerID, containerId),
+                };
+            }
+            catch (Exception ex)
+            {
+                AWSXRayEventSource.Log.ResourceAttributesExtractException(nameof(AWSEKSResourceDetector), ex);
+            }
+
+            return resourceAttributes;
+        }
+
+        private string GetEKSClusterName()
+        {
+            var credentials = this.GetEKSCredentials();
+
+            if (!this.IsEKSProcess(credentials))
+            {
+                return null;
+            }
+
+            var clusterInfo = this.GetEKSClusterInfo(credentials);
+            var clusterInfoObject = this.DeserializeResponse(clusterInfo);
+
+            return clusterInfoObject.Data?.ClusterName;
+        }
+
+        private bool IsEKSProcess(string credentials)
+        {
+            var httpClientHandler = this.CreateHttpClientHandler();
+            var awsAuth = ResourceDetectorUtils.SendOutRequest(AWSAuthUrl, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).Result;
+            return string.IsNullOrEmpty(awsAuth);
```

Comment:
If this is a EKS process, I believe `awsAuth` will not be a null or empty string. In that case, this statement should be negated to return `true` right?

Reply:
I think this is added to ensure that `awsAuth` is valid so that we don't need to run the rest of code if it's not.

Reply:
What is the value of `awsAuth` variable if this is indeed an EKS process? And what is its value otherwise?

Reply:
`aws-auth` is a metadata file, not a variable. It contains some information regarding the role or something. See https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html